[Elixir] New concept exercise: `processes` (#1654)
exercism/v3: languages/elixir/exercises/concept/processes/.docs/introduction.md

> In Elixir, all code runs inside processes.

Reviewer: My compliments for how well you've explained things in this document! Really excellent!

By default, a function will execute in the same process from which it was called. When you need to explicitly run a certain function in a new process, use `spawn/1`:

```elixir
spawn(fn -> 2 + 2 end)
# => #PID<0.125.0>
```

`spawn/1` creates a new process that executes the given function and returns a _process identifier_ (PID). The new process will stay alive as long as the function executes, and then silently exit.

Elixir's processes should not be confused with operating system processes. Elixir's processes use much less memory and CPU. It's perfectly fine to have Elixir applications that run hundreds of Elixir processes.

## Messages

> Processes do not directly share information with one another. The only way to get data in and out of a running process is by _sending messages_.

Reviewer 2: ```suggestion
Processes do not directly share information with one another. Processes _send and receive messages_ to share data.
```
